### PR TITLE
Bug 1480878 - Monitor the health of Push connector job processing

### PR DESCRIPTION
**Pull request by mars-f**
Add Datadog metrics monitoring for the Push job scheduler queue and the
job scheduler queue backlogs for each Push connector.

### CODE REVIEW

**Comment by mars-f** [comment]
@globau could you please check my work against the first user story in the [monitoring requirements doc](https://docs.google.com/document/d/11vX-JnhQJXkXM7Y3ijqU8Z9mYwxou4OMabJi6od6iG8/edit?ts=5b48bc27#heading=h.wf0adxyj462z)?  I found the wording of the user story to be ambiguous, and I wasn't sure if I should measure the queue of all scheduled jobs in the system or just the queue of scheduled jobs for phabbugz.

**Comment by globau** [comment]
> I wasn't sure if I should measure the queue of all scheduled jobs in the system or just the queue of scheduled jobs for phabbugz.

monitoring all enabled connectors is required, so what you've done looks great to me.

File: extensions/Push/lib/Push.pm [hunk 1]
```diff
@@ -50,12 +39,49 @@ sub start {
...
+sub heartbeat {
+    my ($self) = @_;
+    my $dd = Bugzilla->datadog('bugzilla.pushd');
+
+    $dd->gauge('scheduled_jobs', Bugzilla->dbh->selectrow_array('SELECT COUNT(*) FROM push'));
+
+    foreach my $connector ($self->connectors->list) {
+        if ($connector->enabled) {
+            my $lcname = lc $connector->name;
+            $dd->gauge("${lcname}.backlog", Bugzilla->dbh->selectrow_array('SELECT COUNT(*) FROM push_backlog WHERE connector = ?', $connector->name));
```
Comment (dylanwh) [inline extensions/Push/lib/Push.pm]: The second argument of selectrow_array() needs to be undef if you're using placeholders.

```perl
Bugzilla->dbh->selectrow_array('SELECT COUNT(*) FROM push_backlog WHERE connector = ?', undef, $connector->name)
```

**Comment by dylanwh** [comment]
I added logging, and I know the code is being executed as a result, but `bugzilla.pushd.file.backlog' never shows up on datadog. Perhaps we can only to three levels of dots?

**Comment by dylanwh** [comment]
nevermind, it shows up now. Weeeird.

**Pull request merged**